pkg/globalinit/global_initializers.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package globalinit is used to execute different initializers of the CLI
+// based on their specified triggers.  This can be used to cleanup some data
+// or perform some actions based on certain conditions.
+package globalinit
+
+import (
+	"fmt"
+	"io"
+
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+type initializer struct {
```

Comment:
a suggestion: how about supporting providing a name for each initializer registered? It can be useful for a few things:
1. Used in aggregated error list to provide some consistency in the output
2. In the future, should it happen that a particular initializer consistently fails for a CLI user (intentionally caused by the customer or otherwise), it is much easier to provide some means to turn off certain initializers if we can refer to them by name